util/chunk/row_container_test.go

```diff
@@ -102,54 +102,56 @@ func (r *rowContainerTestSuite) TestSpillAction(c *check.C) {
 	tracker = rc.GetMemTracker()
 	tracker.SetBytesLimit(chk.MemoryUsage() + 1)
 	tracker.FallbackOldAndSetNewAction(rc.ActionSpillForTest())
-	c.Assert(rc.AlreadySpilledSafeForTest(), check.Equals, false)
+	require.False(t, rc.AlreadySpilledSafeForTest())
 	err = rc.Add(chk)
 	rc.actionSpill.WaitForTest()
-	c.Assert(err, check.IsNil)
-	c.Assert(rc.AlreadySpilledSafeForTest(), check.Equals, false)
-	c.Assert(rc.GetMemTracker().BytesConsumed(), check.Equals, chk.MemoryUsage())
+	require.NoError(t, err)
+	require.False(t, rc.AlreadySpilledSafeForTest())
+	require.Equal(t, chk.MemoryUsage(), rc.GetMemTracker().BytesConsumed())
 	// The following line is erroneous, since chk is already handled by rc, Add it again causes duplicated memory usage account.
 	// It is only for test of spill, do not double-add a chunk elsewhere.
 	err = rc.Add(chk)
 	rc.actionSpill.WaitForTest()
-	c.Assert(err, check.IsNil)
-	c.Assert(rc.AlreadySpilledSafeForTest(), check.Equals, true)
+	require.NoError(t, err)
+	require.True(t, rc.AlreadySpilledSafeForTest())
 
 	// Read
 	resChk, err := rc.GetChunk(0)
-	c.Assert(err, check.IsNil)
-	c.Assert(resChk.NumRows(), check.Equals, chk.NumRows())
+	require.NoError(t, err)
+	require.Equal(t, chk.NumRows(), resChk.NumRows())
 	for rowIdx := 0; rowIdx < resChk.NumRows(); rowIdx++ {
-		c.Assert(resChk.GetRow(rowIdx).GetDatumRow(fields), check.DeepEquals, chk.GetRow(rowIdx).GetDatumRow(fields))
+		require.Equal(t, chk.GetRow(rowIdx).GetDatumRow(fields), resChk.GetRow(rowIdx).GetDatumRow(fields))
 	}
 	// Write again
 	err = rc.Add(chk)
 	rc.actionSpill.WaitForTest()
-	c.Assert(err, check.IsNil)
-	c.Assert(rc.AlreadySpilledSafeForTest(), check.Equals, true)
+	require.NoError(t, err)
+	require.True(t, rc.AlreadySpilledSafeForTest())
 
 	// Read
 	resChk, err = rc.GetChunk(2)
-	c.Assert(err, check.IsNil)
-	c.Assert(resChk.NumRows(), check.Equals, chk.NumRows())
+	require.NoError(t, err)
+	require.Equal(t, chk.NumRows(), resChk.NumRows())
 	for rowIdx := 0; rowIdx < resChk.NumRows(); rowIdx++ {
-		c.Assert(resChk.GetRow(rowIdx).GetDatumRow(fields), check.DeepEquals, chk.GetRow(rowIdx).GetDatumRow(fields))
+		require.Equal(t, chk.GetRow(rowIdx).GetDatumRow(fields), resChk.GetRow(rowIdx).GetDatumRow(fields))
 	}
 
 	err = rc.Reset()
-	c.Assert(err, check.IsNil)
+	require.NoError(t, err)
 }
 
-func (r *rowContainerTestSerialSuite) TestSpillActionDeadLock(c *check.C) {
+func TestSpillActionDeadLock(t *testing.T) {
+	t.Parallel()
```

Comment:
Serial test, plz don't parallel it.

Reply:
What makes it serial?

Reply:
rowContainerTestSerialSuite, in this test, it is failpoint.

Reply:
Thank you. Resolved.